pkg/conditions/replicasready/handler.go

```diff
@@ -0,0 +1,80 @@
+package replicasready
+
+import (
+	"context"
+
+	"github.com/giantswarm/conditions/pkg/conditions"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	capiexp "sigs.k8s.io/cluster-api/exp/api/v1alpha3"
+	ctrl "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/conditions-handler/pkg/internal"
+	"github.com/giantswarm/conditions-handler/pkg/key"
+)
+
+type HandlerConfig struct {
+	CtrlClient ctrl.Client
+	Logger     micrologger.Logger
+
+	Name         string
+	UpdateStatus bool
+}
+
+type Handler struct {
+	ctrlClient      ctrl.Client
+	internalHandler *internal.Handler
+	logger          micrologger.Logger
+	name            string
+}
+
+func NewHandler(config HandlerConfig) (*Handler, error) {
+	h := &Handler{
+		ctrlClient: config.CtrlClient,
+		logger:     config.Logger,
+		name:       config.Name,
+	}
+
+	internalHandlerConfig := internal.HandlerConfig{
+		CtrlClient:        config.CtrlClient,
+		Logger:            config.Logger,
+		UpdateStatus:      config.UpdateStatus,
+		ConditionType:     capiexp.ReplicasReadyCondition,
+		EnsureCreatedFunc: h.ensureCreated,
+	}
+
+	internalHandler, err := internal.NewHandler(internalHandlerConfig)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	h.internalHandler = internalHandler
+
+	return h, nil
+}
+
+func (h *Handler) EnsureCreated(ctx context.Context, object interface{}) error {
+	machinePool, err := key.ToMachinePoolPointer(object)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return h.internalHandler.EnsureCreated(ctx, machinePool)
+}
+
+func (h *Handler) EnsureDeleted(_ context.Context, _ interface{}) error {
+	return nil
+}
+
+func (h *Handler) Name() string {
+	return h.name
+}
+
+func (h *Handler) ensureCreated(ctx context.Context, object conditions.Object) error {
```

Comment:
This private function is named the same way as a public one.
Isn't this confusing? Do we do this normally?
Plus I don't see where this function is called

Reply:
It's like that in all conditions handlers in this module. Now when we have more of them and more patterns have surfaced, I agree it can be done in a better way and have some ideas in mind, but I would do that in another pull request and keep scope here related to only `ReplicasReady` condition addition.